### ISSUE TITLE
dyld: Library not loaded :

### ISSUE THREAD

**Issue by ganeshbasker** [CocoaPods/CocoaPods#8612]
1. Created a cocoa touch framework with cocoa pods as a dependencies.
2. The framework that I am building is a mixed of swift and Objective C 
3. Even the cocoa pods dependencies are having the mixed codes.
4. While consuming this framework in a sample application, we are getting the below error with crash.

**"dyld: Library not loaded: @rpath/KissXML.framework/KissXML
  Referenced from: /Users/719428/Library/Developer/CoreSimulator/Devices/CE52EC21-4EC9-4403-820C-7782304F0270/data/Containers/Bundle/Application/5E363578-B139-4090-A075-271E38A2F341/SampleEpubReader.app/Frameworks/SampleEpubFramework.framework/SampleEpubFramework
  Reason: image not found"**

There are few more queries as part of this framework development

1. Does cocoa pods support cocoa touch framework as a dependency.
2. In the middle of development, we are facing the issue "Could not find the module "module Name" in pod. Do you have any idea how to resolve this issue. Note :- Temporarily we have the moved the pod module as a source

<img width="331" alt="ProjectSkeleton" src="https://user-images.githubusercontent.com/42533038/54438737-4a901d00-475d-11e9-81e5-ebf7ddee77e7.png">

**Comment by axnmax** [comment]
I also faced this issue. While building a cocoa touch framework that includes some Cocoapod dependencies. The CocoaPod dependencies are not embedded to the framework binary created by Xcode.  Although this works if I make a iOS application all the Cocoapod dependencies are embedded in the .app file as separate .frameworks.

Is there any issue with using Cocoapods and CocoaTouch frameworks together? Is there any workaround for this issue ? 

Please suggest

**Comment by wellbranding** [comment]
Hello, our issue might solve your case:
https://github.com/CocoaPods/CocoaPods/issues/8594#issue-418674487
 we ourselves are struggling with optional frameworks, which do not work with cocoa pods...

**Comment by TomVanDerSpek** [comment]
We are facing the same issue with Cocoa Touch Frameworks. 
```dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /Users/.../Build/Products/Debug-iphonesimulator/Data.framework/Data
  Reason: image not found
```

Podfile that used to work on CocoaPod version 1.5.3. But is broken on 1.6.1:
```#App target
target 'App' do
  pod 'EasyPeasy'
end

# Cocoa Touch Framework
target 'Data' do
  pod 'Alamofire'
end
```
And this seems to "fix" the error:
```#App target
target 'App' do
  pod 'EasyPeasy'
  pod 'Alamofire'
end

# Cocoa Touch Framework
target 'Data' do
  pod 'Alamofire'
end
```

Hope this is fixed in the next update

**Comment by jtreanor** [comment]
There appears to be two issues being discussed here. I am experiencing the same issue as @ganeshbasker and @axnmax when updating to any version of CocoaPods later than `1.6.0.beta.2`.

The issue seems to affect only some pods. So far I am aware of `KissXML` and `GoogleSignInRepacked`. These pods use `vendored_frameworks` to link a dynamic framework but I think there may be something unusual about how they are built. In the cause of [GoogleSignInRepacked](https://github.com/wordpress-mobile/GoogleSignInRepacked) it is actually a [repackaged version](https://pewpewthespells.com/blog/convert_static_to_dynamic.html) of the static framework distributed by Google.

I have put together a sample project to demonstrate the issue here: https://github.com/jtreanor/CocoaPodsLinkingDemo

The change seems to have been introduced by https://github.com/CocoaPods/CocoaPods/pull/8314.

**Comment by bradhilton** [comment]
I'm experiencing the same issue. @TomVanDerSpek 's workaround does seem to work for me.

**Comment by CanYouHelpMe** [comment]
with the latest pods which version is CocoaPods 1.9.0.beta.2,this issue happened,how can i do?

**Comment by hamoonjm** [comment]
The workaround is not a perfect solution.
If you have a SwiftUI file in the framework, and try to use canvas, it'll throw the exact error and basically you can't use the canvas.

**Comment by grago** [comment]
Hello, i'm having the same issue. I've created a Framework library with a CocoaPods dependency.


```use_frameworks!
workspace 'MyWorkspace.xcworkspace'

target 'MyLibrary' do
    project './MyLibrary/MyLibrary.xcodeproj' 
    pod 'WebRTC'
end```

When i run the main app which links to my library, i get the error "dld: Library not loaded. Reason: image not found"

Only if i add the same pod to the main app target, i make the error go away, which seems redundant. What am i doing wrong?

**Comment by adnanafridi619** [comment]
I am also facing same issue . How to fix that issue ?

**Comment by swapnizzle** [comment]
I encountered this issue when trying to build to my actual iOS device (but it worked fine in the simulator). I spent at least 10 hours trying to fix this bug. Tried everything in all the forums, nothing worked. But this is what finally worked for me!

- Upgrading from iOS 13.3.1 to iOS 13.4.1

I couldn't have imagined this, but its true that <b> iOS 13.3.1 breaks the third parties signature unless you have a paid developer account!</b>
